### ISSUE TITLE
Implement L1 and L2 gradient as hooks with the option of changing the weight decay value.

### ISSUE THREAD

**Issue by veritas9872** [pytorch/pytorch#91073]
### 🚀 The feature, motivation and pitch

The L1 and L2 weight decay are important regularizations in deep learning, but their current implementation in PyTorch is not as flexible as one may desire. The L2 weight decay is available in PyTorch optimizers, but the weight decay value cannot be modified during training. L1 is not implemented in an efficient manner in the PyTorch library.
I propose implementing the L1 and L2 weight decay as gradient hooks and placing them in the `torch.nn.utils` namespace. Other regularizations, such as weight and spectral normalization, are already implemented there, as are pruning utilities, which often go with L1 regularization.
The implementation should allow easy attachment and detachment of the hook on any `nn.Module` with options to separate parameter groups. The weight decay value should be modifiable to allow weight decay scheduling.

### Alternatives

Implementing in the Optimizer base class is an option but has failed to gain traction.
Currently, the TorchLayers library offers a Keras-like interface with the L1 and L2 regularizations, but this requires a wrapper class and cannot separate parameter groups very efficiently or change the weight decay value.

### Additional context

I have already implemented the hooks and am willing to contribute. Separation of parameter groups has already been implemented, but I am considering a better interface for exposing the weight decay value for scheduling.
